### PR TITLE
feat: add next-tinacms-cloudinary package to monorepo

### PR DESCRIPTION
**Pull request by jbevis**
Adds the ```next-tinacms-cloudinary``` package to the monorepo.

Very much a WIP, and we can use this PR for dialogue regarding documentation and best use cases.

### CODE REVIEW

**Comment by jbevis** [comment]
This is a capital "D" draft. The demo is currently not working, and it will require some though as it was conceived to work with Tina Cloud, so there may need to be some adjusting to handle only a TinaCMS use case.

Comments/thoughts welcome.

**Comment by logan-anderson** [comment]
Ran, a bootstrap and then a build and got this error.

```
Circular dependency: ../../../../../node_modules/util/util.js -> ../../../../../node_modules/util/node_modules/inherits/inherits.js -> ../../../../../node_modules/util/util.js
Circular dependency: ../../../../../node_modules/util/util.js -> ../../../../../node_modules/util/node_modules/inherits/inherits.js -> /Users/logananderson-forestry-mac/node_modules/util/util.js?commonjs-proxy -> ../../../../../node_modules/util/util.js
Circular dependency: ../../node_modules/util/util.js -> ../../node_modules/util/node_modules/inherits/inherits.js -> ../../node_modules/util/util.js
Circular dependency: ../../node_modules/util/util.js -> ../../node_modules/util/node_modules/inherits/inherits.js -> /Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/util.js?commonjs-proxy -> ../../node_modules/util/util.js
/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/@tinacms/scripts/bin/scripts.js:35
  throw err
  ^

Error: 'promisify' is not exported by ../../node_modules/util/util.js
    at error (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:10151:30)
    at Module.error (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:14137:9)
    at handleMissingExport (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:14056:21)
    at Module.traceVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:14448:17)
    at ModuleScope.findVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:13146:39)
    at FunctionScope.findVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:2981:38)
    at ChildScope.findVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:2981:38)
    at FunctionScope.findVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:2981:38)
    at ChildScope.findVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:2981:38)
    at FunctionScope.findVariable (/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/rollup/dist/rollup.js:2981:38) {
  code: 'MISSING_EXPORT',
  url: 'https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module',
  pos: 761,
  loc: {
    column: 9,
    file: '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts',
    line: 19
  },
  frame: "17: import { NextApiRequest, NextApiResponse } from 'next'\n" +
    "18: import multer from 'multer'\n" +
    "19: import { promisify } from 'util'\n" +
    '             ^\n' +
    '20: \n' +
    '21: export interface CloudinaryConfig {',
  watchFiles: [
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/next-tinacms-cloudinary/src/index.ts',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/next-tinacms-cloudinary/src/handlers/index.ts',
    '\x00tslib',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/packages/next-tinacms-cloudinary/src/handlers/sign-signature-handler.ts',
    '/Users/logananderson-forestry-mac/node_modules/path/path.js',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/util.js',
    '/Users/logananderson-forestry-mac/node_modules/util/util.js',
    '\x00commonjsHelpers.js',
    '\x00/Users/logananderson-forestry-mac/node_modules/util/util.js?commonjs-proxy',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/support/isBuffer.js',
    '/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/node_modules/inherits/inherits.js',
    '\x00/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/support/isBuffer.js?commonjs-proxy',
    '\x00/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/node_modules/inherits/inherits.js?commonjs-proxy',
    '/Users/logananderson-forestry-mac/node_modules/util/support/isBuffer.js',
    '/Users/logananderson-forestry-mac/node_modules/util/node_modules/inherits/inherits.js',
    '\x00/Users/logananderson-forestry-mac/node_modules/util/support/isBuffer.js?commonjs-proxy',
    '\x00/Users/logananderson-forestry-mac/node_modules/util/node_modules/inherits/inherits.js?commonjs-proxy',
    '/Users/logananderson-forestry-mac/node_modules/util/node_modules/inherits/inherits_browser.js',
    '\x00/Users/logananderson-forestry-mac/node_modules/util/node_modules/inherits/inherits_browser.js?commonjs-proxy',
    '\x00/Users/logananderson-forestry-mac/dev/tinacms/tinacms/node_modules/util/util.js?commonjs-proxy'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! next-tinacms-cloudinary@0.41.1-alpha.1 build: `tinacms-scripts build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the next-tinacms-cloudinary@0.41.1-alpha.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/logananderson-forestry-mac/.npm/_logs/2021-06-15T12_41_46_160Z-debug.log

lerna ERR! npm run build exited 1 in 'next-tinacms-cloudinary'
```

Looks like there is some dependency stuff that needs to be fixed?

**Comment by jbevis** [comment]
@logan-anderson I was running into a bunch of dep issues too. try running ```npm run hard-reset```. That's what got it running for me.

File: packages/demo-next/pages/_app.js [hunk 1]
```diff
@@ -48,11 +49,14 @@ export default class Site extends App {
...
       },
+      media: new CloudinaryMediaStore(
+        process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME
+      ),
       //plugins: [CustomPaginatorPlugin],
     })
     const client = new GitClient('http://localhost:3000/___tina')
     this.cms.registerApi('git', client)
-    this.cms.media.store = new NextGitMediaStore(client)
+    // this.cms.media.store = new NextGitMediaStore(client)
```
Comment (Jaza) [inline packages/demo-next/pages/_app.js:59]: Did you mean to commit this?

File: packages/next-tinacms-cloudinary/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
```
Comment (Jaza) [inline packages/next-tinacms-cloudinary/CHANGELOG.md:5]: Shouldn't this file only start out with the intro lines up to here, and nothing else?

File: packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    // TODO: Fix only one file upload
+    const { file, directory } = media[0]
+    const formData = new FormData()
+
+    formData.append('file', file)
+    formData.append('directory', directory)
+    formData.append('filename', file.name)
+
+    const res = await fetch(`/api/cloudinary/media`, {
```
Comment (Jaza) [inline packages/next-tinacms-cloudinary/src/cloudinary-media-store.ts:46]: Would probably be better to move the actual Cloudinary API calls into code outside of the media store class, and/or into a `CloudinaryClient` class, for more modular code, and for consistency with the other media stores.

**Pull request closed without merging**